tests/test_converter.py

```diff
@@ -1110,3 +1114,49 @@ def cost(x, y):
         ]
 
         assert np.allclose(jac, jac_expected)
+
+    def test_meas_circuit_in_qnode(self, qubit_device_2_wires):
```

Comment:
Can we add a test that specifically targets the behaviour of distinguishing between a mid-circuit and a terminal measurement? For example this circuit:

```
qc = QuantumCircuit(3, 3)
qc.h(0)
qc.measure(0, 0) 
qc.rx(0.2, [0])
qc.cx(0, 1)
qc.measure(1, 1)
```

versus this one:

```
qc = QuantumCircuit(3, 3)
qc.h(0)
qc.measure(3, 3) 
qc.rx(0.2, [0])
qc.cx(0, 1)
qc.measure(1, 1)
```

Reply:
Also, it seems for Qiskit that `qc.measure` goes to a classical register, where `qc.measure_all()` goes to a separate readout register unless you tell it to read out to the primary classical register. Does that influence how we interpret whether they are mid-circuit or terminal measurements? Just wondering if terminal measurements should be measurements read out to the `meas` register rather than being based on whether additional operations are performed. Not sure if this is for this PR or a followup PR regarding mid-circuit measurements, or if it even makes sense to consider. @trbromley